### ISSUE TITLE
More efficient shapes and paths

### ISSUE THREAD

**Issue by alvaromontoro** [CSS-Next/logo.css#27]
This may be unnecessary optimiziation, but is there a reason why the file has such complex and long paths? Especially the container, which is a simple rectangle with 3 rounded corners:

`<path fill="#639" d="M0 0H840C882.435 0 923.131 16.8571 953.137 46.8629C983.143 76.8687 1000 117.565 1000 160V840C1000 882.435 983.143 923.131 953.137 953.137C923.131 983.143 882.435 1000 840 1000H160C117.565 1000 76.8687 983.143 46.8629 953.137C16.8571 923.131 0 882.435 0 840V0Z"/>`

Those 284 bytes could be reduced to 169 using a simpler path (which in my opinion would be easier to read and understand too):

`<path fill="#639" d="M0,0 C0,0 0,0 840,0 928,0 1000,72 1000,160 1000,160 1000,840 1000,840 1000,928 928,1000 840,1000 840,1000 160,1000 160,1000 72,1000 0,928 0,840Z"/>`

And if we are not attached to a single path method, we could even simplify it more (to 102 bytes) using two rectangles, one with a radius, and the other one without:

`<rect fill="#639" width="1000" height="1000" rx="160"/><rect fill="#f008" width="300" height="300"/>`

And using groups, reduced it to 95 characters:

`<g fill="#639"><rect width="1000" height="1000" rx="160"/><rect width="300" height="300"/></g>`

From 284 to 95 is a 66% size reduction, only on that path. I haven't tested the letters, but they would offer some optimization too (although I doubt a 66%).

----

As I was writing the issue, I passed the SVG file through the [SVG OMG](https://jakearchibald.github.io/svgomg/) tool, and the file size goes down from 2.78KB to 1.88KB

```
<svg xmlns="http://www.w3.org/2000/svg" width="1000" height="1000">
  <path fill="#639" d="M0 0h840a160 160 0 0 1 160 160v680a159.999 159.999 0 0 1-160 160H160A159.999 159.999 0 0 1 0 840V0Z"/>
  <path fill="#fff" d="M817.5 923.1c-32.3 0-57-9.4-74.3-28.3-17.1-19-26-46.1-26.8-81.5H786c.4 31.3 11.4 47 33 47 11 0 18.8-3.5 23.3-10.4 4.4-6.9 6.7-17.9 6.7-33 0-12-3-22-8.8-29.9a73 73 0 0 0-29.4-21.1L785 753.5c-23-11-39.7-24-50-39.2-10-15.1-15-36.3-15-63.4 0-30.3 8.8-54.9 26.3-73.8 18-18.9 42.5-28.3 73.8-28.3 30.2 0 53.6 9.3 70.1 27.8 16.8 18.6 25.6 45.2 26.3 80h-67a72.011 72.011 0 0 0-6.7-33c-3.8-7.6-11.2-11.4-22.2-11.4-19.6 0-29.4 11.7-29.4 35 0 11.8 2.4 21 7.2 28a65.002 65.002 0 0 0 25.3 17.4l24.7 10.9c25.5 11.7 43.9 26 55.2 43.3 11.4 17.2 17 40.2 17 69 0 34.5-9 61-27.3 79.5-18.2 18.5-43.5 27.8-75.8 27.8Zm-224.8 0c-32.3 0-57-9.4-74.3-28.3-17.1-19-26-46.1-26.8-81.5h69.6c.4 31.3 11.4 47 33 47 11 0 18.8-3.5 23.2-10.4 4.5-6.9 6.7-17.9 6.7-33 0-12-2.9-22-8.7-29.9a73 73 0 0 0-29.4-21.1l-25.8-12.4c-23-11-39.7-24-50-39.2-10-15.1-15-36.3-15-63.4 0-30.3 8.8-54.9 26.3-73.8 18-18.9 42.5-28.3 73.8-28.3 30.2 0 53.6 9.3 70 27.8 17 18.6 25.7 45.2 26.4 80h-67a72.011 72.011 0 0 0-6.7-33c-3.8-7.6-11.2-11.4-22.2-11.4-19.6 0-29.4 11.7-29.4 35 0 11.8 2.4 21 7.2 28a65.002 65.002 0 0 0 25.3 17.4l24.7 10.9c25.5 11.7 43.9 26 55.2 43.3 11.3 17.2 17 40.2 17 69 0 34.5-9.1 61-27.3 79.5-18.2 18.5-43.5 27.8-75.8 27.8Zm-234.1 0c-31.6 0-56.7-8.6-75.3-25.8-18.6-17.1-27.8-42.8-27.8-76.8v-168c0-33.8 9.8-59.4 29.4-77 19.9-17.4 46-26.2 78.3-26.2 31.3 0 56 9 74.3 26.8 18.5 17.9 27.8 44.5 27.8 80v13.3h-73.7v-12.9c0-13.7-3-23.8-8.8-30.4a26.397 26.397 0 0 0-21.6-10.3c-12 0-20.5 4.1-25.3 12.4a78.99 78.99 0 0 0-6.7 36.6v146c0 30.5 10.7 46 32 46.3 10 0 17.5-3.6 22.7-10.8a51.7 51.7 0 0 0 7.7-30.4V805h73.7v11.4c0 23.7-4.6 43.5-13.9 59.3a88.597 88.597 0 0 1-38.1 35.6 121.002 121.002 0 0 1-54.7 11.8Z"/>
</svg>
```

**Comment by Que-tin** [comment]
The SVGs are currently only exported, with no further optimizations. Your suggestions make sense and look good to me, I'll look into them later and see if these can be applied upon all of our base logos and have no further effect on our asset generation.

If you want to you could also open a PR and I'll check on it later.

**Comment by myfonj** [comment]
I've considered using rounded `rect` square and "masking" its left corner with sharp one as well, but subjectively decided that introducing extra elements could be possibly problematic. (Objectively I see no problem with that, just a gut feeling.)

But it is fact that using such complicated cubic Béziers only to get rounded corners feels wrong, especially in a logo of CSS where "elliptical rounding" is more established. Simpler arcs could do the same with better precision. Coincidentally filed as #28 for that.

**Comment by argyleink** [comment]
the svg was exported from figma and created with a large canvas, then it was ran through SVGOMG before uploaded to this repo. aka, there's definitely room to optimize this file and i super welcome it ❤️

**Comment by rol4nd909** [comment]
I opened a pull request for this one #35

**Comment by mxdvl** [comment]
I've got something in the works that is super highly optimised which I'll be ready to share in a day or two… beats SVGO by a wide margin 😉

**Comment by alvaromontoro** [comment]
@mxdvl That sounds exciting. Does it involve creating a path for the S and referencing it?

**Comment by rol4nd909** [comment]
Closed by #35

**Comment by mxdvl** [comment]
For anyone pondering on whether there's any room to squeeze the rounded rectangle in the background, I made [6 nearly visually identical variations using different techniques](https://codepen.io/mxdvl/pen/XWvLEPj). They mostly all fit within a few characters of each other. I believe the current approach of using a `path` with proper arcs is the best option, and while it's possible to get a slightly smaller file size it would come at a cost on readability, ease of editing or mathematical approximations with Béziers curves.
